### PR TITLE
Chat: Remove Report Button from user's own profile sidebar

### LINKED ISSUE #4589: Self-Report on DEV Connect
**Describe the bug**
Users are able to report themselves for abuse in DEV Connect. Clicking the button will lead to a report for the user on the other end of the chat. 

**To Reproduce**
1. Enter a private message with another user in DEV Connect. 
2. Click on your own username to open the profile sidebar. 
3. Click on the report button. 

**Expected behavior**
Users should not have a report button in their own profile sidebar, just like they don't have a block button. 

**Screenshots**
selfreport1
selfreport2
When you do report yourself for abuse, it will link the user or group you were speaking to. In this case, I was speaking to Michael when I found the bug so it linked to him. 

**Desktop (please complete the following information):**

- OS: Windows 10
- Browser: Chrome 
- Version: 77.0.3865.120

### PR DESCRIPTION
**Pull request by bolariin**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Documentation Update

## Description
Currently a user can self report themselves via user profile in the sidebar. This removes the report button from a user's sidebar when they look at their own profile.

## Related Tickets & Documents
Resolves #4589 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
Sidebar when looking at own profile

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

File: app/javascript/chat/userDetails.jsx [hunk 1]
```diff
@@ -89,12 +89,21 @@ export default class UserDetails extends Component {
...
-          btnName: 'Block User'
+      blockButton = setUpButton({
+        modalId: 'userdetails__blockmsg',
+        otherModalId: 'userdetails__reportabuse',
+        btnName: 'Block User',
       });
     }
+
+    let reportButton = '';
+    // if (window.currentUser.id !== user.id) {
```
Comment (rhymes) [inline app/javascript/chat/userDetails.jsx]: shouldn't this be part of the code?

Comment (bolariin) [reply app/javascript/chat/userDetails.jsx]: Yes, it should. I forgot to uncomment it

Comment (rhymes) [reply app/javascript/chat/userDetails.jsx]: can you add a test for it as well? Thanks!

**Comment by bolariin** [comment]
Yeah, I’ll take a look at what’s going on

**Pull request merged**